### DISCUSSION TITLE
Make operations take less clicks

### DISCUSSION THREAD

**Discussion by justpeanuts** [gsantner/markor#2103] [Ideas]
Markor has a particular niche amongst note taking applications in being an excellent note app that is  particularly SNAPPY in opening the app and thereby being able to quickly get to making notes. But having to do many unnecessary taps/clicks to do basic operations defeats this purpose and makes things tedious -
1. Creating a new note with a particular template
Currently when creating a new note , one has to do the following steps 
(Click 0 - opening the app)
Click 1 - the "+" icon
Click 2 - the template drop down menu
Click 3 - select the zettelkasten template
Click 4 - the "OK" button 
Click 2 and 3 can be eliminated by having an option in settings to choose a different template other than a blank file.
Click 4 is tedious because the OK button is in the middle of the screen when the users hands are at the bottom of the screen near the keyboard - this hand movement can be eliminated if the enter key in the keyboard doubled as the OK button for the dialogue.

2. To take a photo with the camera and link it to the note
Click 1 - the " attach" tool icon
Click 2 - "image" option
Click 3 - "camera picture" option
Click 4 - camera app's shutter button
Click 5 - OK button
If the user wants to add say 2 more photos then it would take a total of 15 clicks!

If instead of putting all the different attach options under a single "attach" button , if the "take photo from camera" was separated as a separate tool in toolbar then click 1 and 2 can be eliminated. Click 5 is mostly unnecessary as most people wouldn't always want to edit the filename of the image (if they do they can always navigate back and rename the file manually). Another separate toolbar tool to add a batch of photos at once would also help.

**Comment by harshad1** [comment]
FWIW I am working on making the attach system a for images / recordings etc a little better. Will push up a PR soon. Goal is to make things <= 3 clicks.

**Reply by justpeanuts** [reply]
Thank you Harshad!

**Comment by harshad1** [comment]
@justpeanuts Please look at #2106 

I would appreciate if you could test this (install the debug version from https://github.com/gsantner/markor/actions/runs/6139542187?pr=2106)

**Reply by harshad1** [reply]
Note: It can be installed side-by-side with the main app

**Reply by justpeanuts** [reply]
Hi, I tested it, it works well! My only comments on it are:
i) the 3 clicks that are required now could be minimised to just two clicks if the "insert image from camera" and "insert image from filesystem" were made into two separate toolbar options.
ii) actually, in effect, it requires a fourth click, because keyboard does not open by default after inserting the image link. It would be great if keyboard popped up with the cursor being at new line by having a line-break after the image-link.
iii) the filename of the image taken from the camera is like this : IMG_20230912-152630.jpg5470420763913322814.jpg - nothing wrong, just seemed a little weird.
iv) it would be superb if there was another "batch-image-add" option to take multiple images form the camera and markor automatically makes a pdf out of it and attaches the link to the pdf. (But i understand that that would probably be way too much work) 
v) "Marder" really..? you gave me a scare there.. lol.

Anyhow, thank you very much for implementing it!
